### ISSUE TITLE
"Crash" on launch in 1.37.1

### ISSUE THREAD

**Issue by MarcelloNicoletti** [microsoft/vscode#79256]
- VSCode Version: 1.37.1 (I think, I can't open the about window but the new version page came up)
- OS Version: macOS 10.14.6

Steps to Reproduce:

1. Launch Visual Studio Code after an auto-update to 1.37.1
2. Notice app open indicator in dock blink on and then off
3. A VSC window opens and persists even though macOS thinks VSC is not running (dock indicator is still gone)

Within this window I've been able to do everything I've tried that only uses the mouse. macOS doesn't consider the window focused so keyboard input is sent to whichever the most recent other app to have focus was. Using the mouse I was able to select text, open workspaces from the welcome page, switch editor tabs, and create and save new files. All while macOS still believes VSC is closed.

Does this issue occur when all extensions are disabled?: Yes

**Comment by MarcelloNicoletti** [comment]
Since the window doesn't gain focus VSC's menu bar is always hidden. I was able to get to Code's menu bar by toggling full screen, meaning no other apps in the workspace could be focused. Using the menu bar I was able to get to the Issue reporter

Issue Type: <b>Bug</b>

"Crash" on launch in 1.37.1

VS Code version: Code 1.37.1 (f06011ac164ae4dc8e753a3fe7f9549844d15e35, 2019-08-15T16:16:34.800Z)
OS version: Darwin x64 18.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4578U CPU @ 3.00GHz (4 x 3000)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (7.87GB free)|
|Process Argv|/Users/marcello/test_file|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|10.5.0
bracket-pair-colorizer|Coe|1.0.61
vscode-markdownlint|Dav|0.30.1
vscode-eslint|dba|1.9.0
gitlens|eam|9.9.3
vscode-html-css|ecm|0.2.0
vscode-npm-script|eg2|0.3.8
prettier-vscode|esb|1.9.0
sass-lint|gle|1.0.5
todo-tree|Gru|0.0.141
rainbow-csv|mec|1.2.0
vscode-language-babel|mgm|0.0.23
debugger-for-chrome|msj|4.11.7
sass-indented|rob|1.5.1
code-settings-sync|Sha|3.4.1
vscode-nginx|sha|0.6.0
vscode-icons|vsc|9.2.0

(1 theme extensions excluded)

</details>


NOTE: I changed the file listed in the argv table cell. The actual name isn't important. But it is the same path that I launched code with several launch attempts ago. So it seems like VSC never really fully closed since that first time on the new version

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by MarcelloNicoletti** [comment]
After restarting my machine I can't reproduce it with the regular build. I'm not entirely sure how I got it into that state. Fully shutting down cleared whatever it was though. Sorry that I can't give more info. I did try launching the insider's app but wasn't able to get this issue to occur either.

**Comment by MarcelloNicoletti** [comment]
I recreated this issue recently. It isn't a VSCode bug. It has to do with tmux sessions and old user namespaces in macOS. When trying to diagnose an Applescript issue I came across [this article](http://www.economyofeffort.com/2013/07/29/reattach-to-user-namespace-the-fix-for-your-tmux-in-os-x-woes/) which includes the following symptom.

> * Launching a GUI app from the terminal and getting a “ghost window”: the app window loads in the background, with no dock icon, cannot be Command-Tab’d to, and the app’s menu does not populate the top bar when the window gains focus.

This matches what I was reporting here. I can recreate it about half the time with the following steps:

1. Log in to a fresh user session in macOS
2. Make sure there is no tmux server running (`tmux kill-server`)
3. Start a new tmux session (`tmux new -t "Test"`)
4. Start VSCode from within tmux using the `code` command
5. Logout of the macOS session
6. Wait some time and then log back in with the same user account
7. Attach to the tmux session started in step 3 (`tmux attach -t "Test"`)
8. Repeat step 4

After step 7 all kinds of things are broken beyond just VSCode. I can't copy or paste using `pbcopy`, Applescript commands against any application fail, and other apps launched using `open -A <name>` also exhibit this same odd window behavior. More time spent between steps 5 and 6 makes this more likely. This will probably be fixed by using [`reattach-to-user-namespace`](https://github.com/ChrisJohnsen/tmux-MacOSX-pasteboard) in my tmux configuration on macOS.

I'm commenting for the benefit of anyone who might stumble onto this issue from search.
